src/rgw/rgw_auth.cc

```diff
@@ -363,10 +363,15 @@ void rgw::auth::WebIdentityApplier::modify_request_state(const DoutPrefixProvide
   s->info.args.append("sub", token_claims.sub);
   s->info.args.append("aud", token_claims.aud);
   s->info.args.append("provider_id", token_claims.iss);
+  s->info.args.append("client_id", token_claims.client_id);
 
   string idp_url = get_idp_url();
   string condition = idp_url + ":app_id";
-  s->env.emplace(condition, token_claims.aud);
+  if (! token_claims.client_id.empty()) {
```

Comment:
According to AWS document, the trust policy of the role should have a condition where app_id must the same as the value of 'aud' field of the token and the value must be the client id. This condition establishes trust between a particular application and a role. The current version of Keycloak doesn't set its aud field to client id (older versions did), so I have modified the condition to be set to the client id (client id is a different field in the token) if present else set it to aud. Is this Ok? Or should the condition be always set to the value of 'aud' field irrespective of whether 'aud' is contains client id or not.

Reply:
It /sounds/ equivalent?  Maybe both aud and clientid are relevant?  Marcus, Adam, can you give any insight into these fields and what it means to condition on one or the other?